app/Console/Commands/BillingCalculate.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Models\Bill;
+use App\Models\BillHistory;
+use Illuminate\Console\Command;
+use LibreNMS\Util\Number;
+
+require_once base_path() . '/includes/init.php';
+
+class BillingCalculate extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'billing:calculate {--r}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'calculate the port usage for the billing.';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        if ($this->option('r')) {
+            $this->info("Clearing history table.\n");
+            BillHistory::truncate();
+        }
+        //get all bills and order by bill_id
+        foreach (Bill::orderBy('bill_id')->get() as $bill) {
+            $this->info(str_pad($bill->bill_id . ' ' . $bill->bill_name, 30) . " \n");
+            $i = 0;
+            while ($i <= 24) {
+                unset($class);
+                unset($rate_data);
+                $day_data = getDates($bill->bill_day, $i);
+
+                $datefrom = $day_data['0'];
+                $dateto = $day_data['1'];
+                $check = BillHistory::where(['bill_id' => $bill->bill_id, 'bill_datefrom' => $datefrom, 'bill_dateto' => $dateto])->first();
+                $period = getPeriod($bill['bill_id'], $datefrom, $dateto);
```

Comment:
You can port these functions to class based ones, and if they are used in other places just replace the original function with a `return myNewFunction()`

we done that in many places.

Reply:
This exact function probably should be in a model tho